src/classes/RD2_ScheduleService.cls

```diff
@@ -698,12 +695,11 @@ public without sharing class RD2_ScheduleService {
 
             if (isScheduleUpdateNeeded(rd, oldRd)) {
                 if (triggerAction == TDTM_Runnable.Action.BeforeUpdate &&
-                    rd.StartDate__c == oldRd.StartDate__c &&
-                    rd.Status__c != RD2_Constants.STATUS_CLOSED
-                ) {
+                        rd.StartDate__c == null &&
```

Comment:
Is there ever a scenario where rd.StartDate can be null? I believe this is required field not just in the UI, but also in our validation triggers.

Reply:
Without this code, the following code results in a null pointer exception:

`npe03__Recurring_Donation__c rd = new npe03__Recurring_Donation__c(Id='a09J000000O12Q5IAJ', npe03__Amount__c = 400, StartDate__c = null);
update rd;
`

Reply:
I see your point. I think we only require on the page layout, but not in our apex validation.